api/PclusterApiHandler.py

```diff
@@ -696,34 +696,30 @@ def _get_params(_request):
 
 
 class PclusterApiHandler(Resource):
-    method_decorators = [authenticated({"admin"})]
+    __read_decorators = [authenticated({'admin'})]
+    __write_decorators = [authenticated({'admin'}), csrf_needed]
+
+    method_decorators = {
+        'get': __read_decorators,
```

Comment:
issue: are we sure about this? What's the drawback of checking CSRF on `GET`s as well? 

The fact that `GET`s are non-mutating is not guaranteed unless we trust the way APIs are implemented

Reply:
From my tests it seems that GET calls are actually to read data and not change them (I may have missed something of course)
we could check csrf for GETs as well if we wanted to, I'm open to it